File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.java
class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        int[] answer = new int[2];
        
        int x_max = board[0] / 2;
        int x_min = 0 - (board[0] / 2);
        int y_max = board[1] / 2;
        int y_min = 0 - (board[1] / 2);
        
        for(int i = 0; i < keyinput.length; i++){
            switch(keyinput[i]){
                case "left":
                    if(answer[0] != x_min) answer[0]--;
                    break;
                case "right":
                    if(answer[0] != x_max) answer[0]++;
                    break;
                case "up":
                    if(answer[1] != y_max) answer[1]++;
                    break;
                case "down":
                    if(answer[1] != y_min) answer[1]--;
                    break;
            }
        }

        return answer;
    }
}

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    answer = [0, 0]
    xy = [board[0] // 2, board[1] // 2]
    
    for m in keyinput:
        if m == "up":
            if answer[1] != xy[1]:
                answer[1] += 1
        elif m == "down":
            if answer[1] != -xy[1]:
                answer[1] -= 1
        elif m == "left":
            if answer[0] != -xy[0]:
                answer[0] -= 1
        elif m == "right":
            if answer[0] != xy[0]:
                answer[0] += 1
    
    return answer